humility-core/src/hubris.rs

```diff
@@ -570,9 +580,60 @@ impl HubrisArchive {
             qualified_variables: MultiMap::new(),
             unions: HashMap::new(),
             definitions: MultiMap::new(),
+            namespaces: Vec::new(),
         })
     }
 
+    //
+    // For a namespace identifer, compose and return the entire namespace
+    // vector as strings.
+    //
+    fn namespace(&self, namespace: Option<usize>) -> Result<Vec<&String>> {
+        let mut rval = vec![];
+        let mut current = namespace;
+
+        while let Some(namespace) = current {
+            if namespace >= self.namespaces.len() {
+                bail!("namespace id {namespace} is invalid");
+            }
+
+            let (name, parent) = &self.namespaces[namespace];
+            rval.push(name);
```

Comment:
I'd suggest making this function return `Result<Vec<&str>>`, so this would become
```suggestion
            rval.push(name.as_str());
```